app/src/main/kotlin/de/janhopp/luebeckmensawidget/api/MensaApi.kt

```diff
@@ -8,24 +8,18 @@ import de.janhopp.luebeckmensawidget.utils.toMensaApiFormat
 import io.ktor.client.*
 import io.ktor.client.call.*
 import io.ktor.client.request.*
-import kotlinx.serialization.json.Json
 import kotlin.coroutines.cancellation.CancellationException
 
 class MensaApi(
     private val client: HttpClient = HttpClient(),
 ) {
-    private val json = Json {
-        ignoreUnknownKeys = true
-        isLenient = true
-    }
-
     private val urlDay: String
         get() = currentTime.mensaDay.dayOfWeek.toMensaApiFormat()
 
     suspend fun getMealsToday(): List<MensaDay> = runCatching {
         client.get("https://speiseplan.mcloud.digital/meals?day=$urlDay")
             .body<String>()
-            .let { days -> json.decodeFromString<List<MensaDay>>(days) }
+            .let { days -> MensaJson.decodeFromString<List<MensaDay>>(days) }
```

Comment:
Here you only get one day back from the API, but the variable is called "days". And you also save it later via the function "setMensaDays()".

Reply:
I think it would make sense to load several days in advance. After all, it doesn't need much memory.